### DISCUSSION TITLE
Multiple date picker with ranges on shift key

### DISCUSSION THREAD

**Discussion by TheSisb** [gpbl/react-day-picker#2263] [Support]
Hi, thank you for the wonderful library!

I'm trying to build a datepicker that defaults to the `mode="multiple"` behavior, but allows for ranges to be selected if the shift key is held.  

I've examined the shift key example in the repo however it isn't working as expected; modifying the `Day` component does not preserve the out-of-month styling and kind of seems like the wrong approach maybe:
<img width="94" alt="image" src="https://github.com/user-attachments/assets/6552e49a-8ed7-401c-9126-9b70611b1c60">

I've found this issue where the concept of multiple ranges were discussed but it also seems like a half answer: https://github.com/gpbl/react-day-picker/issues/92

I'd be satisfied with a solution that selects each date individually between two dates if `shift` is held, so `mode="multiple"`'s state can be preserved. 

Any advice is appreciated! Thanks

**Comment by gpbl** [comment]
@TheSisb are you on DayPicker 8? Please give a look to the [upcoming release](https://daypicker.dev/next) as it may help with your issue with the custom component style.

Your use case is a bit difficult to implement, but not impossible: check out this updated example and see if it is of inspiration for you:

https://daypicker.dev/next/guides/custom-components#example-range-with-shift-key
